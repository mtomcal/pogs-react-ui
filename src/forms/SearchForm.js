import React from 'react';
import PropTypes from 'prop-types';
import { Field, reduxForm } from 'redux-form';
import RenderField from './RenderField';
import validator from 'validator';
import _ from 'lodash';

function validate(values) {
  let errors = {};
  const keysOfInterest = _.keys(values)
      .filter((key) => !_.isEmpty(values[key]));

  keysOfInterest.forEach((key) => {
      if (key === 'gene' && !validator.isAlphanumeric(values[key])) {
          errors.gene = 'Gene Id should be alphanumeric';
      }
      if (key === 'keyword' && !validator.isAlphanumeric(values[key])) {
          errors.keyword = 'Keyword should be alphanumeric';
      }
      if (key === 'pog' && !validator.isInt(_.get(values, 'pog', ''))) {
          errors.pog = 'POG Id should be an integer';
      }
  });
  return errors;
}

function SearchForm({submitting, handleSubmit}) {
  return (
      <form onSubmit={handleSubmit}>
        <div className="row">
          <div className="col-sm-12">
            <Field label="Search By Gene ID e.g. GRMZM2G055768" name="gene" type="text" component={RenderField} />
          </div>
        </div>
        <div className="row">
          <div className="col-sm-12">
            <Field label="Search By Keyword e.g. 'ppr' or by InterproID e.g. IPR001117" name="keyword" type="text" component={RenderField} />
          </div>
        </div>
        <div className="row">
          <div className="col-sm-12">
            <Field label="Search By POG ID i.e. 5562" name="pog" type="text" component={RenderField} />
          </div>
        </div>
        <div className="row">
          <div className="col-sm-12">
            <button className="btn btn-primary btn-raised" type="submit" disabled={submitting}>Submit</button>
          </div>
        </div>
      </form>
  );
}

SearchForm.propTypes = {
  handleSubmit: PropTypes.func,
  reset: PropTypes.func
};

export default reduxForm({
  form: 'search',
  validate
})(SearchForm);
