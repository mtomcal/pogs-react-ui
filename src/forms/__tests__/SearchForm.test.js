import React from 'react';
import SearchForm from '../SearchForm';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import {mount} from 'enzyme';
import state from '../../State';

describe('SearchForm', () => {
  test('renders default form correctly', () => {
    const component = renderer.create(
      <Provider store={state}>
        <SearchForm />
      </Provider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('renders gene field', () => {
    const component = mount(
      <Provider store={state}>
        <SearchForm />
      </Provider>
    );
    expect(component.find('input[name="gene"]').exists()).toEqual(true);
  });
  test('renders keyword field', () => {
    const component = mount(
      <Provider store={state}>
        <SearchForm />
      </Provider>
    );
    expect(component.find('input[name="keyword"]').exists()).toEqual(true);
  });
  test('renders pog field', () => {
    const component = mount(
      <Provider store={state}>
        <SearchForm />
      </Provider>
    );
    expect(component.find('input[name="pog"]').exists()).toEqual(true);
  });
});



