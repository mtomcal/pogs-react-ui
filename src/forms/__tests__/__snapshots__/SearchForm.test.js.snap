exports[`SearchForm renders default form correctly 1`] = `
<form
  onSubmit={[Function]}>
  <div
    className="row">
    <div
      className="col-sm-12">
      <div
        className="form-group"
        onBlur={[Function]}
        onFocus={[Function]}>
        <label
          className="control-label"
          htmlFor="gene">
          Search By Gene ID e.g. GRMZM2G055768
        </label>
        <input
          className="form-control"
          name="gene"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          type="text"
          value="" />
        <span />
      </div>
    </div>
  </div>
  <div
    className="row">
    <div
      className="col-sm-12">
      <div
        className="form-group"
        onBlur={[Function]}
        onFocus={[Function]}>
        <label
          className="control-label"
          htmlFor="keyword">
          Search By Keyword e.g. \'ppr\' or by InterproID e.g. IPR001117
        </label>
        <input
          className="form-control"
          name="keyword"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          type="text"
          value="" />
        <span />
      </div>
    </div>
  </div>
  <div
    className="row">
    <div
      className="col-sm-12">
      <div
        className="form-group"
        onBlur={[Function]}
        onFocus={[Function]}>
        <label
          className="control-label"
          htmlFor="pog">
          Search By POG ID i.e. 5562
        </label>
        <input
          className="form-control"
          name="pog"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          type="text"
          value="" />
        <span />
      </div>
    </div>
  </div>
  <div
    className="row">
    <div
      className="col-sm-12">
      <button
        className="btn btn-primary btn-raised"
        disabled={false}
        type="submit">
        Submit
      </button>
    </div>
  </div>
</form>
`;
